### ISSUE TITLE
Custom game path ending with `<storeUserId>` includes `<storeUserId>`'s parent directory

### ISSUE THREAD

**Issue by redactedscribe** [mtkennerly/ludusavi#236]
### Ludusavi version

v0.19.0

### Operating system

Windows

### Installation method

Scoop

### Description

I've added a single path for a custom game, a game that doesn't exist in the manifest (i.e. my custom game entry doesn't override anything), and the path I've defined ends in `.../<storeUserId>`. The path I'm trying to add ends in a folder that is 17 numeric characters, so I assume that this is a `<storeUserId>`.

When previewing this path in the _Backup Mode_ tab, the files found are those under `<storeUserId>` _and_ those in `<storeUserId>`'s parent (i.e. items at the same hierarchy level as `<storeUserId>`). Only files under `<storeUserId>` should be detected by the preview.

Changing the custom path to be the game data's literal path (i.e. with the actual 17 numbers), then Ludusavi correctly detects files within that path, and only that path.

Thanks.

**Comment by mtkennerly** [comment]
Related discussion: https://github.com/mtkennerly/ludusavi/issues/39

Right now, `<storeUserId>` is regrettably just interpreted as `*`, because I'm not sure if it's feasible to account for all possible ID patterns. For example, [one comment](https://github.com/mtkennerly/ludusavi/issues/39#issuecomment-815860719) in the other ticket noted that, within a Steam root, GTA 4 used some other pattern for its ID rather than a typical Steam pattern.

Maybe we could have an option to enable stricter matching of known patterns, but there will be a risk of missing saves. For the default configuration, I think it's better to find too many files than too few, since at least it gives the user a chance to deselect them instead of miss them entirely.

**Comment by redactedscribe** [comment]
> Right now, `<storeUserId>` is regrettably just interpreted as `*`

This shouldn't mean that the files outside of the path should be matched, right?

For a hierarchy of `/a/b/` with custom path `/a/<storeUserId>/` i.e. `/a/*/`, only items in `/a/b/` should be matched. Currently items inside `/a` are being matched. I see the issue now:  I was presuming `a/*/` with a trailing slash would only match folders, but it probably does make sense to match any items, not just folders, with or without a trailing slash.

> Maybe we could have an option to enable stricter matching of known patterns, but there will be a risk of missing saves.

Maybe the solution to varying `<storeUserId>` values would be to have a strict set of known patterns by default, and when a pattern can't be unambiguously matched, allow the user to pick which folder it is _on backup_ and for Ludusavi to remember that value going forward. If the tracked value changes, ask the user to pick again. If a value was needed to be tracked by Ludusavi, then _on restore_, if the path already exists on the system, use that, else ask to pick a different folder as the stand-in for `<storeUserId>`. If this idea would better as alternate behaviour, have a toggle to enable strict behaviour for those that want to backup exactly as specified by the manifest. This was just my first idea, maybe there's a better way.

**Comment by mtkennerly** [comment]
> I see the issue now: I was presuming `a/*/` with a trailing slash would only match folders, but it probably does make sense to match any items, not just folders, with or without a trailing slash.

Yeah, Ludusavi strips trailing slashes, mainly for the non-wildcard case since `/a` and `/a/` will normally find the same results. I could see an argument that it shouldn't strip the slash when preceded by `*`, but if you really do just want to match in subfolders, then you can specify `/a/*/*` instead.

I should probably add a README section that covers this.

> when a pattern can't be unambiguously matched, allow the user to pick which folder it is

The tricky thing is that, if we can't find the path, is it because we have the wrong ID pattern or just because the game isn't installed? We don't want to end up bugging the user about a bunch of games they've never even played. We could narrow it down to cases where we found *some* saves for a game but no matches for a `<storeUserId>` path, but then we also need to be very careful about not warning for potentially irrelevant paths (e.g., for a different OS or store). We might also have false positives if a game removes a `<storeUserId>` file on uninstall, but leaves some others.

It's hard 😅 I can definitely add an opt-in set of known patterns, but the alerting is much more complex. Also, because of how, for example, Steam games can require other launchers or account systems with their own user IDs, it will probably need to check all patterns for all root types (excluding `other`, which would continue using `*`).
